theano/gof/cmodule.py

```diff
@@ -415,9 +415,13 @@ def get_module_hash(src_code, key):
             # libraries to link against.
             to_hash += list(key_element)
         elif isinstance(key_element, string_types):
-            if key_element.startswith('md5:'):
-                # This is the md5 hash of the config options. We can stop
-                # here.
+            if (key_element.startswith('md5:') or
+                    key_element.startswith('sha256:') or
+                    key_element.startswith('hash:')):
+                # This is actually a sha256 hash of the config options.
+                # Currently, we still keep md5 to don't break old Theano.
+                # We add 'sha256:' and 'hash:' so that when we change it in
+                # the futur, it won't break this version of Theano.
```

Comment:
The hash split the key in 2 part, the "soft" part and the "hard" part. I don't want to change that logic now. This would go in the refactoring of the cache.

I'm not to remove from the code the possibility to do "sha256:" in the futur. If we want to add the name, we could do "hash:NAME:" and it will always work as we only check for "hash:"